src/pages/signup-affiliates/components/account-details.tsx

```diff
@@ -0,0 +1,349 @@
+import React, { useEffect, useState } from 'react'
+import { WizardStepProps } from '../_types'
+import affiliate_validation from '../validations/_affilaite_validation'
+import BirthForm from '../utils/_birth-form'
+import AffiliateInput from '../utils/_affiliate-input'
+import AffiliatesHeader, { InputGroup, InputWrapper } from '../utils/_affiliate-header'
+import { localize } from 'components/localization'
+
+const AccountDetails = ({
+    is_individual,
+    affiliate_account,
+    updateData,
+    onValidate,
+}: WizardStepProps) => {
+    const affiliate_data = affiliate_account.personal_details
+    const [username, setUsername] = useState(affiliate_data.username)
```

Comment:
There is a lot of individual state items here, and then a switch statement below to handle the editing of these state items.

How would you feel about doing this instead:

```
const [formData, setFormData] = useState({});
```

Then in your `handleInput` you can:

```
const handleInput = (e) => {
    const { name, value } = e.target;
    const validationFunction = affiliate_validation[name];

    setFormData(prevFormData => ({
        ...prevFormData,
        [name]: value
    }));

    if (validationFunction) {
        const errorMsg = validationFunction(value);
        setFormData(prevFormData => ({
            ...prevFormData,
            [`${name}_error_msg`]: errorMsg
        }));
    }
};
```

Reply:
thanks for the very useful comment @markwylde-deriv 🙏 State already changes with function like managment